app/models/energy/ScheduleBase.py

```diff
@@ -0,0 +1,60 @@
+"""Schedule Base Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+
+
+class UnitType (str, Enum):
+    dimensionless = 'Dimensionless'
+    temperature = 'Temperature'
+    delta_temperature = 'DeltaTemperature'
+    precipitation_rate = 'PrecipitationRate'
+    angle = 'Angle'
+    convection_coefficient = 'ConvectionCoefficient'
+    activity_level = 'ActivityLevel'
+    velocity =  'Velocity'
+    capacity =  'Capacity'
+    power =  'Power'
+    availability =  'Availability'
+    percent = 'Percent'
+    control = 'Control'
+    mode = 'Mode'
+
+class DateTime(BaseModel):
+    """DateTime."""
+
+    month: int = Schema(
```

Comment:
No year?

Reply:
I was using the DateTime class in the pollination schema as reference https://github.com/pollination/simulation-service/blob/master/app/models/common/datetime.py. Should I add the year as well?

Reply:
I suggest moving this definition to the `app/models/common/datetime.py` file.  I would suggest giving it a year, but you do want to stay consistent with the rest of the schema.  Maybe needs `is_leap_year` too

Reply:
@mostaphaRoudsari and I recently had a conversation and realized that we will need different types of schedules for daylight and energy (largely because of E+ limitations with schedules). While I know I originally said we wanted a common datetime, I think we want to keep date and time as separate objects implemented only on energy as @tanushree04 has it implemented now.  Let's just make sure that the type of input is coordinated with how we have other datetimes in ladybug/honeybee, meaning that time starts counting from hour 0:00 to 23:59 (not 1 to 24).

Reply:
Update: Minute should still not be able to go past 59 but it seems we will have to support hours going to 24 unless we plan to completely break with E+ convention.

Reply:
This implementation is missing `is_leap_year`. We need to do the validation for leap year in date level otherwise you have to do it in several places.